hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/downloader/NodeSignatureVerifier.java

```
@@ -126,17 +126,29 @@ private boolean verifySignature(FileStreamSignature fileStreamSignature,
             return false;
         }
 
-        if (fileStreamSignature.getSignature() == null) {
+        if (fileStreamSignature.getEntireFilesignature() == null) {
             log.error("Missing signature data: {}", fileStreamSignature);
             return false;
         }
 
         try {
             log.trace("Verifying signature: {}", fileStreamSignature);
-            Signature sig = Signature.getInstance("SHA384withRSA", "SunRsaSign");
+            Signature sig;
+            if (fileStreamSignature.getSignatureType() == SignatureType.SHA_384_WITH_RSA) {
+                sig = Signature.getInstance("SHA384withRSA", "SunRsaSign");
+            } else {
+                log.error("Signature type not valid: {}", fileStreamSignature);
+                return false;
+            }
             sig.initVerify(publicKey);
-            sig.update(fileStreamSignature.getHash());
-            return sig.verify(fileStreamSignature.getSignature());
+            sig.update(fileStreamSignature.getEntireFileHash());
+            boolean isMatch = sig.verify(fileStreamSignature.getEntireFilesignature());
```

fail early when entire file hash signature verification fails

---

Good call, refactored to fail fast.